js/src/util/subscribe-to-events.spec.js

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { spy, stub } from 'sinon';
+
+import subscribeToEvents from './subscribe-to-events';
+import {
+  pastLogs, liveLogs, createApi, createContract
+} from './subscribe-to-events.test.js';
+
+const delay = (t) => new Promise((resolve) => {
+  setTimeout(resolve, t);
+});
+
+describe('util/subscribe-to-events', () => {
+  beforeEach(function () {
+    this.api = createApi();
+    this.contract = createContract(this.api);
+  });
+
+  it('installs a filter', async function () {
+    const { api, contract } = this;
+
+    subscribeToEvents(contract, [ 'Foo', 'Bar' ]);
+    await delay(0);
```

Comment:
Thought about that as well. Decided for `await delay(9)` (which is somewhat random tbh) because using fake timers introduces potential errors & bugs by manipulating the global timing functions.

Have no real opinion on this, will adapt to whatever you and @ngotchac prefer.